arroyo-worker/src/connectors/kinesis/sink/mod.rs

```diff
@@ -0,0 +1,238 @@
+use std::{
+    marker::PhantomData,
+    time::{Duration, SystemTime},
+};
+
+use anyhow::Result;
+use arroyo_macro::{process_fn, StreamNode};
+use arroyo_rpc::OperatorConfig;
+use arroyo_types::{CheckpointBarrier, Data, Key, Record};
+use aws_config::load_from_env;
+use aws_sdk_kinesis::{
+    client::fluent_builders::PutRecords, model::PutRecordsRequestEntry, types::Blob,
+    Client as KinesisClient,
+};
+use serde::Serialize;
+use tracing::warn;
+use uuid::Uuid;
+
+use crate::engine::Context;
+
+use super::{KinesisTable, TableType};
+
+#[derive(StreamNode)]
+pub struct KinesisSinkFunc<K: Key + Serialize, T: Data + Serialize> {
+    client: Option<KinesisClient>,
+    in_progress_batch: Option<BatchRecordPreparer>,
+    flush_config: FlushConfig,
+    name: String,
+    _phantom: PhantomData<(K, T)>,
+}
+
+#[process_fn(in_k = K, in_t = T, tick_ms=10)]
+impl<K: Key + Serialize, T: Data + Serialize> KinesisSinkFunc<K, T> {
+    pub fn from_config(config: &str) -> Self {
+        let config: OperatorConfig =
+            serde_json::from_str(config).expect("Invalid config for KafkaSink");
+        let table: KinesisTable =
+            serde_json::from_value(config.table).expect("Invalid table config for KafkaSource");
+        let flush_config = FlushConfig::new_from_table(&table);
+        Self {
+            client: None,
+            in_progress_batch: None,
+            name: table.stream_name,
+            flush_config,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn name(&self) -> String {
+        format!("kinesis-producer-{}", self.name)
+    }
+
+    async fn on_start(&mut self, _ctx: &mut Context<(), ()>) {
+        let config = load_from_env().await;
```

Comment:
what do you think about allowing users to configure the kinesis endpoint? This would allow local development (with something like kinesalite) and would allow talking to kinesis in different regions.

Reply:
I think we should handle that when we build out more complete AWS configuration support. Right now we mainly assume the default environment will tell you what you need.